src/main/kotlin/com/vk/admstorm/services/HastebinService.kt

```diff
@@ -3,17 +3,35 @@ package com.vk.admstorm.services
 import com.intellij.openapi.components.Service
 import com.intellij.openapi.components.service
 import com.intellij.openapi.project.Project
-import com.vk.admstorm.CommandRunner
 import com.vk.admstorm.env.Env
+import com.vk.admstorm.env.getByKey
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import java.net.URI
+import java.net.http.HttpClient
+import java.net.http.HttpResponse
 
 @Service(Service.Level.PROJECT)
-class HastebinService(private val project: Project) {
+class HastebinService {
     companion object {
         fun getInstance(project: Project) = project.service<HastebinService>()
     }
 
     fun createHaste(data: String): String {
         val output = data.replace("\"", "\\\"").replace("$", "\\$")
-        return CommandRunner.runRemotely(project, "echo \"$output\" | ${Env.data.pasteBinCommand}").stdout
+        val hastLink = Env.data.services.getByKey("hastebin")?.url ?: return ""
+
+        val client = HttpClient.newBuilder().build()
+        val request = java.net.http.HttpRequest.newBuilder().uri(URI.create("$hastLink/documents"))
+            .POST(java.net.http.HttpRequest.BodyPublishers.ofString(output)).build()
+        val response = client.send(request, HttpResponse.BodyHandlers.ofString())
+
+        val jsonResponse = Json.parseToJsonElement(response.body())
+        if (jsonResponse is JsonObject) {
+            val value = jsonResponse["key"]?.jsonPrimitive?.content
+            return "$hastLink/${value}"
+        }
+        return ""
```

Comment:
why not `null`?

Reply:
That's a good question. In this case, you can get by with an empty string, since otherwise you will have to write wrappers everywhere to check for null. In general, it is possible to make String!

Reply:
Yes, it is necessary to process this case everywhere in a good way

Reply:
Replaced